source/setup.Rmd

```diff
@@ -1,282 +1,277 @@
+  to help you with this, as editing the BIOS can be dangerous. Detailed instructions for doing this are beyond the scope of this book. 
+More information on installing and running
+Docker on Windows, as well as troubleshooting tips, can be found [in the online
+Docker documentation](https://docs.docker.com/desktop/install/windows-install/).
+
+**Running JupyterLab** Once Docker is running (in either Windows or MacOS), you need to download and run the  
+Docker *image* that we have made available for the worksheets (an *image* is like a "snapshot" of a 
+computer with all the right packages pre-installed). In the search bar, enter `ubcdsci/r-dsci-100`, as this is 
+the name of the image. You will be presented
+with a few options (Figure \@ref(fig:docker-desktop-search)). In the "Tag" drop down menu, click 
+the tag `20230724142640d8f06f` to select the correct version of the image. Then click 
+the "Pull" button to download the image.
+
+```{r docker-desktop-search, echo = FALSE, fig.cap = "The Docker Desktop search window. Make sure to click the Tag drop down menu and find the right version of the image before clicking the Pull button to download it.", fig.retina = 2, out.width="100%"}
+image_read("img/setup/docker-1.png") |> 
+  image_crop("3632x2000")
+```
 
-**MacOS:** To install miniconda on MacOS, you will need to use a different 
-installation method depending on the type of processor chip your computer has.
+Once the image is done downloading, click the "Images" button on the left side of the Docker Desktop window (Figure \@ref(fig:docker-desktop-images)). You
+will see the recently downloaded image listed there under the "Local" tab.
 
-If your Mac computer has an Intel x86 processor chip you can download 
-the [latest Python 64-bit version from here](https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.pkg). 
-After the download has finished, run the installer and accept the default 
-configuration for all pages.
+```{r docker-desktop-images, echo = FALSE, fig.cap = "The Docker Desktop images tab.", fig.retina = 2, out.width="100%"}
+image_read("img/setup/docker-2.png") |> 
+  image_crop("3632x2000")
+```
 
-If your Mac computer has an Apple M1 processor chip you can download 
-the [latest Python 64-bit version from here](https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-arm64.sh).
-After the download has finished, you need to run the downloaded script in the terminal using a command
-like:
+To start up a *container* using that image, click the play button beside the image. This will open the run configuration menu (Figure \@ref(fig:docker-desktop-runconfig)).
+In the "Host port" textbox, enter `8888`. In the "Volumes" section, click the "Host path" box and navigate to the folder where your Jupyter worksheets
+are stored. In the "Container path" text box, enter `/home/jovyan/work`. Then click the "Run" button to start the container.
```

Comment:
We really should give some recommendations on the following:

1. Where student should store the downloaded worksheets (I would suggest documents)
2. What path to suggest for the Host path. I know in older versions of Docker, Docker has major performance issues if the host path is set to the computers home directory on mac. It essentially becomes unusable... So we want to avoid students from doing that...

Reply:
We also need to provide some shutdown instructions if we want students to use Docker Desktop on Mac and Windows. This is because when you run a container with the UI (as opposed to the command line) it depends how you stop it as to whether you get the behaviour you want that is equivalent to  to `--rm`.  To get the equivalent to `--rm` students have to click the red, not the stop button or the grey garbage can, it has to be the red garbage can.

<img width="1269" alt="Screenshot 2023-08-28 at 12 31 35 PM" src="https://github.com/UBC-DSCI/introduction-to-datascience/assets/6865016/b3d77d6c-c9f7-41eb-b915-559063f55a89">

If they click the stop button, the container it doesn't get deleted and instead still sits there stopped, but using the port allocated to it. This means no new container can take the port until that one is deleted. Alternatively students could just restart the container via Docker Desktop to get up and running again, but we need to teach them how to do that.

Reply:
good catch!